### PR TITLE
Add visually hidden component, export a11y styles

### PR DESCRIPTION
**Pull request by amberleyromo**
- Add a visually hidden component to the library.
  - This applies the `visuallyHiddenCss` style to a `<span>` and wraps children. This is a common and necessary pattern, it makes sense to me to have this defined in our component library. Interested to hear thoughts!
- Export a11y styles (`visuallyHiddenCss` and `resetvisuallyHiddenCss`).
  - I don't really see a reason not to keep this here, and reference wherever needed in docs/dashboard/any of the other properties. (Occurred to me in the splitting of static and dashboard code, cc @mfrachet)

### CODE REVIEW

**Review by fk — APPROVED** [review]
Yay! Happy to help replacing https://github.com/gatsby-inc/mansion/blob/1ab41a3bc5e1f1fdd96e0b51f506e5326c6b82b5/cloud/gatsbyjs.com/src/utils/styles.js#L268-L279!

**Comment by fk** [comment]
Wondering if I'll still be able to do this: https://github.com/gatsby-inc/mansion/blob/1ab41a3bc5e1f1fdd96e0b51f506e5326c6b82b5/cloud/gatsbyjs.com/src/components/blog/meta.js#L42 ?

**Comment by amberleyromo** [comment]
Yep, this PR is also exporting the `visuallyHiddenCss` from `gatsby-interface` (The same styles living in multiple places).

- interface: https://github.com/gatsby-inc/gatsby-interface/blob/main/src/stylesheets/a11y.ts#L3
- mansion: https://github.com/gatsby-inc/mansion/blob/1ab41a3bc5e1f1fdd96e0b51f506e5326c6b82b5/cloud/gatsbyjs.com/src/utils/styles.js#L268

So in the case you linked, you would import that style from gatsby-interface instead: https://github.com/gatsby-inc/mansion/blob/1ab41a3bc5e1f1fdd96e0b51f506e5326c6b82b5/cloud/gatsbyjs.com/src/components/blog/meta.js#L3

File: src/components/VisuallyHidden/VisuallyHidden.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core"
+import { visuallyHiddenCss } from "../../stylesheets/a11y"
+
+export type VisuallyHiddenProps = {
```
Comment (Elanhant) [inline src/components/VisuallyHidden/VisuallyHidden.tsx:5]: We should allow any props supported by `span` for cases where we want to pass an `id` to link the contained text to some other element via `aria-describedby` or `aria-labelledby` (we do this in several places in Cloud).
```typescript
export type VisuallyHiddenProps = React.ComponentPropsWithoutRef<"span">
```

File: src/components/VisuallyHidden/VisuallyHidden.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  children: React.ReactNode
+}
+
+function VisuallyHidden({ children }: VisuallyHiddenProps) {
+  return <span css={visuallyHiddenCss}>{children}</span>
+}
+
+export default VisuallyHidden
```
Comment (Elanhant) [inline src/components/VisuallyHidden/VisuallyHidden.tsx:13]: We should use named exports instead, makes it easier when it comes to searching for a component or re-exporting via index files.

File: src/components/VisuallyHidden/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { VisuallyHiddenProps as VisuallyHiddenPropsDefinition } from "./VisuallyHidden"
```
Comment (Elanhant) [inline src/components/VisuallyHidden/index.ts:1]: Once `VisuallyHidden` is exposed as a named export, we can use a wildcard export here:
```typescript
export * from "./VisuallyHidden"
```

File: src/components/VisuallyHidden/VisuallyHidden.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/** @jsx jsx */
```
[the remark this answers is not in the document]
Comment (amberleyromo) [reply src/components/VisuallyHidden/VisuallyHidden.stories.tsx:1]: My bad, I should have noticed! Maybe we should touch on this in our next FE catchup meeting, for everyone?

[the remark this answers is not in the document]
Comment (Elanhant) [reply src/components/VisuallyHidden/VisuallyHidden.stories.tsx:1]: Great idea! I will add it to the agenda. Hopefully I will also get to work on improving docs during the upcoming sprint.

**Comment by fk** [comment]
> So in the case you linked, you would import that style from gatsby-interface instead

Oh nice, thanks for making old man see! 🤗 
(I _was_ looking for that export when I left my review :D 🤦‍♂️)

**Pull request merged**